src/core/client/admin/components/UserStatus/UserStatusSitesListContainer.tsx

```diff
@@ -0,0 +1,239 @@
+
+interface ScopeSite {
+  readonly id: string;
+  readonly name: string;
+}
+
+export interface Scopes {
+  role: USER_ROLE;
+  sites?: ScopeSite[] | null;
+}
+
+interface Props {
+  viewerScopes: Scopes;
+  userScopes: Scopes;
+}
+
+interface Props {
+  query: UserStatusSitesListContainer_query;
+  relay: RelayPaginationProp;
+
+  viewerScopes: Scopes;
+  userScopes: Scopes;
+}
+
+const siteIsVisible = (
+  id: string,
+  viewerSites: ScopeSite[] | null | undefined
+) => {
+  if (!viewerSites || viewerSites.length === 0) {
+    return true;
+  }
+
+  return viewerSites.map((s) => s.id).includes(id);
+};
+
+const UserStatusSitesListContainer: FunctionComponent<Props> = ({
+  query,
+  relay,
+  viewerScopes,
+}) => {
+  const viewerIsScoped = useMemo(() => {
+    return viewerScopes.sites && viewerScopes.sites.length > 0;
+  }, [viewerScopes]);
+  const [showSites, setShowSites] = useState<boolean>(!!viewerIsScoped);
+
+  const { input: selectedIDsInput } = useField<string[]>("selectedIDs");
+  const sites = useMemo(() => {
+    if (
+      viewerScopes.role === GQLUSER_ROLE.MODERATOR &&
+      viewerScopes.sites &&
+      viewerScopes.sites.length > 0
+    ) {
+      return viewerScopes.sites || [];
+    }
+    return query?.sites.edges.map((edge) => edge.node) || [];
+  }, [query?.sites.edges, viewerScopes.role, viewerScopes.sites]);
+  const [loadMore, isLoadingMore] = useLoadMore(relay, 1);
+  const [, isRefetching] = useRefetch<
+    UserStatusSitesListContainerPaginationQueryVariables
+  >(relay);
+  const loading = !query || isRefetching;
+  const hasMore = !isRefetching && relay.hasMore();
+
+  const toggleShowSites = useCallback(() => {
+    setShowSites(!showSites);
+  }, [showSites, setShowSites]);
+  const onChangeSite = useCallback(
+    (siteID: string) => () => {
+      const changed = [...selectedIDsInput.value];
+
+      const index = changed.indexOf(siteID);
+      if (index >= 0) {
+        changed.splice(index, 1);
+      } else {
+        changed.push(siteID);
+      }
+
+      selectedIDsInput.onChange(changed);
+    },
+    [selectedIDsInput]
+  );
+
+  return (
+    <IntersectionProvider>
+      <FieldSet>
+        <div className={styles.header}>
+          {viewerIsScoped ? (
+            <Localized id="community-banModal-selectSites">
+              <Label>Select sites to ban</Label>
+            </Localized>
+          ) : (
+            <FormField>
+              <CheckBox checked={showSites} onChange={toggleShowSites}>
+                <Localized id="community-banModal-selectSites">
+                  Select sites to ban
+                </Localized>
+              </CheckBox>
+            </FormField>
+          )}
+        </div>
+
+        {(viewerIsScoped || showSites) && (
+          <FormField>
+            <ListGroup>
+              {sites.map((site) => {
+                const enabled = siteIsVisible(site.id, viewerScopes.sites);
+                const isChecked = selectedIDsInput.value.includes(site.id);
+
```

Comment:
Add a filter in a memo before the render to filter out sites that are not visible.